### DISCUSSION TITLE
How to import const from a .d.ts file?

### DISCUSSION THREAD

**Discussion by AnzhiZhang** [nuxt/nuxt#19826] [Questions] [answered]
I have an example code.

https://stackblitz.com/edit/github-ukqiak?file=app.vue

When I try to import a constant from a .d.ts file, there will be a vite issue: "Failed to load url /types/1 (resolved id: /types/1) in /home/projects/github-ukqiak/app.vue. Does the file exist?"

I tried to import from `~/types/1.d.ts` instead of `~/types/1`, this will work but TS told me I cannot import with a .d.ts extension.

My current solution is to put my constants and types in a .ts file, but I want to put types in a .d.ts file.

Is there any solution or am I misunderstanding the .d.ts file?

---

The reason I am putting a constant in the types file is that I want to declare a literal list and use its string list.

```ts
export const MALES = [
    "male",
    "female",
] as const;

export type Male= typeof MALES[number];
```

---

Maybe related: #15986

**Comment by hslee2008** [comment, marked as answer]
`.d.ts` is a Typescript Declaration File.
You cannot use import or export statements in a .d.ts file. 
The purpose of a .d.ts file is to provide type information for JavaScript code that does not have explicit type annotations, and it is not intended to contain any executable code. 
So, only types is allowed.

That is why you cannot import from a file named `.d.ts`

**Reply by AnzhiZhang** [reply]
Thanks
